src/models/keyboard_model.py

```diff
@@ -0,0 +1,87 @@
+
+import codecs
+import re
+
+
+class KeyboardModel:
+    def __init__(self) -> None:
+        pass
+
+    def _create_keyboard(self, matrix, keyboard_name, image_ids):
+        """
+        Creates a .tec file using the matrix and downloaded images.
+        """
+        keyboard_file = f"{keyboard_name}.tec"
+        try:
+            with codecs.open(keyboard_file, "w", "cp1252") as file:
+                for row in matrix:
+                    file.writelines(["LINHA ?\n", "GRUPO ?\n"])
+                    words = self._split_row(row)
+                    for word in words:
+                        if word.startswith("[") and word.endswith("]"):
+                            clean_word = word[1:-1]
+                            if not image_ids:
+                                print(f"Warning: No more image IDs available for '{clean_word}'.")
```

Comment:
Remover os prints, porque na versao final quando o utilizador usar aplicao nao vai ver estes erros, porque nao tem acesso ao terminal. Ou seja depende do que for tens de demonstar esse erro ao Utilizador. Na `main_view.py` ha uma funcao `show_error` que e responavel para mostar os erros ao user.